src/Symfony/Component/BrowserKit/AbstractBrowser.php

```diff
@@ -29,6 +29,9 @@
  * you need to also implement the getScript() method.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @template TRequest of object
+ * @template TResponse of object
```

Comment:
Let's also type the `$request` and `$response` properties accordingly. That way, Psalm can tell us when we break the contract in the future. Also, this can benefit others extending from `AbstractBrowser` (given they are protected).

Reply:
done